cmake/modules/CMakeLists.txt

```diff
@@ -15,6 +15,11 @@ set(llvm_cmake_builddir "${LLVM_BINARY_DIR}/${LLVM_INSTALL_PACKAGE_DIR}")
 get_property(CIRCT_EXPORTS GLOBAL PROPERTY CIRCT_EXPORTS)
 export(TARGETS ${CIRCT_EXPORTS} FILE ${circt_cmake_builddir}/CIRCTTargets.cmake)
 
+get_property(CIRCT_ALL_LIBS GLOBAL PROPERTY CIRCT_ALL_LIBS)
+get_property(CIRCT_DIALECT_LIBS GLOBAL PROPERTY CIRCT_DIALECT_LIBS)
+get_property(CIRCT_CONVERSION_LIBS GLOBAL PROPERTY CIRCT_CONVERSION_LIBS)
+get_property(CIRCT_TRANSLATION_LIBS GLOBAL PROPERTY CIRCT_TRANSLATION_LIBS)
```

Comment:
I don't think you have anything that uses these?

Reply:
These need to be defined in the lexical environment so that they are substituted into `CIRCTConfig.cmake.in` during the call to `configure_file`. I can't find a better way to substitute properties, but I'm no CMake wizard. This is taken from MLIR.

Reply:
Right...  My only concern is that this might encourage people to use the environment variable rather than the property and they might eventually get out of sync.  In MLIR, the property is used as the golden source of truth and only extracted in the places when needed, IIRC.

Reply:
I hope I'm getting this right:

The environment variable I'm creating is only accessible in `cmake/modules/CMakeLists.txt` and any sub-directories (there are none).  If this was in the top level CMakeLists.txt, then I could see there being a problem.  The `GLOBAL` specifier here is to specify that the property should be loaded from global properties (as opposed to a specific target's property).

Reply:
I can separate the stuff related to exporting in to a separate PR.  I did all the work at the same time to make sure that what I was doing worked.  How does that sound?

Reply:
Yes, you're right about the scoping.  I thought this was in the toplevel CMakeLists.txt.  This is fine as it is.